Upgraded dependencies to use latest dp-kafka and dp-net.

DavidSubiros opened the pull request:
### What

- Upgraded dependencies to latest dp-kafka (to use Sarama and drop Sarama-Cluster):
  - Added `KAFKA_VERSION` config, according to the new dp-kafka constructor parameters.
  - This fixes the Kafka blackout issue
- Upgraded other dependencies to use latest dp-net
  - Migrated go-ns/server dependency to dp-net/http

### How to review

- Make sure code changes make sense
- Make sure unit tests pass

### Who can review

Anyone

DavidSubiros commented:
Draft PR, until we release and tag the new dp-kafka changes

nshumoogum reviewed (approved):
Looks fine to me, I assume pr will update once you have tagged a semantic version to dp-kafka so will have to approve pr again.

DavidSubiros commented:
> Looks fine to me, I assume pr will update once you have tagged a semantic version to dp-kafka and updated the dependency in go.mod so will have to approve pr again.

I think it was already tagged and updated in the last commit of this PR (I also had to update imports to use /v2)
